server/src/test/scala/cromwell/engine/WorkflowManagerActorSpec.scala

```diff
@@ -18,5 +25,89 @@ class WorkflowManagerActorSpec extends CromwellTestKitWordSpec with WorkflowDesc
       val actualOutput = salutation.valueString.trim
       actualOutput should endWith("/call-whereami/execution")
     }
+
+    "pick up workflows" in {
+      case class SubWorkflows(naptime: FiniteDuration) extends SampleWdl {
+        override def workflowSource(runtime: String): WorkflowSource = root
+
+        override val rawInputs: ExecutableInputMap = Map("parent.naptime" -> naptime.toSeconds.toInt)
+
+        val root =
+          """
+            |version 1.0
+            |
+            |import "subsub.wdl" as sub
+            |
+            |workflow parent {
+            |  input {
+            |    Int naptime
+            |  }
+            |  scatter (i in range(3)) {
+            |    call sub.sub { input: naptime = naptime }
+            |  }
+            |}
+          """.stripMargin.trim
+
+        val sub =
+          """
+            |version 1.0
+            |
+            |task snooze {
+            |  input {
+            |    Int naptime
+            |  }
+            |  command {
+            |    echo "zzzz"; sleep ~{naptime}
+            |  }
+            |}
+            |
+            |workflow sub {
+            |  input {
+            |    Int naptime
+            |  }
+            |  call snooze { input: naptime = naptime }
+            |}
+            |
+        """.stripMargin.trim
+
+        override val imports: Option[Set[WorkflowImport]] =
+          Option(
+            Set(
+              WorkflowImport(name = "subsub.wdl", content = sub)
+            )
+          )
+      }
+
+      val config = CromwellTestKitSpec.DefaultConfig.
+        withValue("system.max-concurrent-workflows", ConfigValueFactory.fromAnyRef(2)).
+        withValue("system.new-workflow-poll-rate", ConfigValueFactory.fromAnyRef(1))
+
+      val rootActor = buildCromwellRootActor(config)
+      val serviceRegistryActor = rootActor.underlyingActor.serviceRegistryActor
+
+      val firstSources = SubWorkflows(naptime = 60 seconds).asWorkflowSources()
+
+      def waitForState(workflowId: WorkflowId, state: WorkflowState): Unit = {
+        eventually { verifyWorkflowState(serviceRegistryActor, workflowId, state) } (config = defaultPatience, pos = implicitly[org.scalactic.source.Position])
+      }
+
+      val firstWorkflowId = rootActor.underlyingActor.submitWorkflow(firstSources)
+      waitForState(firstWorkflowId, WorkflowRunning)
+      // Give the subworkflows plenty of time to start running.
+      Thread.sleep(10 * 1000L)
+
+      // Submit a second workflow.
+      val secondSources = SubWorkflows(naptime = 0 seconds).asWorkflowSources()
+      val secondWorkflowId = rootActor.underlyingActor.submitWorkflow(secondSources)
+      // Wait a long time so that there would have been many root workflow pickup sweeps.
+      Thread.sleep(30 * 1000L)
```

Comment:
ToL: I think we could also check that the workflow store actor is receiving a `FetchRunnableWorkflows` message with a `0` value. But now that I'm writing this I think your way is more black box-y which is probably better.